freeipa/src/main/java/com/sequenceiq/freeipa/service/freeipa/user/UmsUsersStateProvider.java

```diff
@@ -64,23 +64,27 @@
             Map<String, FmsGroup> crnToFmsGroup = grpcUmsClient.listGroups(actorCrn, accountId, List.of(), requestIdOptional).stream()
                     .collect(Collectors.toMap(Group::getCrn, this::umsGroupToGroup));
 
-            List<WorkloadAdministrationGroup> wags = grpcUmsClient.listWorkloadAdministrationGroups(INTERNAL_ACTOR_CRN, accountId, requestIdOptional);
-
+            Map<WorkloadAdministrationGroup, FmsGroup> wags = grpcUmsClient.listWorkloadAdministrationGroups(INTERNAL_ACTOR_CRN, accountId, requestIdOptional)
+                    .stream()
+                    .collect(Collectors.toMap(wag -> wag, wag -> nameToGroup(wag.getWorkloadAdministrationGroupName())));
             environmentCrns.forEach(environmentCrn -> {
-                UmsUsersState.Builder umsUsersStateBuilder = new UmsUsersState.Builder();
+                UmsUsersState.Builder umsUsersStateBuilder = new UmsUsersState.Builder()
+                        .setWorkloadAdministrationGroups(wags.values());
+
                 UsersState.Builder usersStateBuilder = new UsersState.Builder();
                 Set<String> wagNamesForOtherEnvironments = new HashSet<>();
 
+
                 crnToFmsGroup.values().forEach(usersStateBuilder::addGroup);
 
                 // Only add workload admin groups that belong to this environment.
                 // At the same time, build a set of workload admin groups that are
                 // associated with other environments so we can filter these out in
                 // the per-user group listing in handleUser.
-                wags.forEach(wag -> {
+                wags.keySet().forEach(wag -> {
```

Comment:
I think you could iterate the entryset, so you won't have to make a get() against the map